### PR TITLE
Extracted UrlManagerInterface from yii\web\UrlManager

### PR DESCRIPTION
**Pull request by njasm**
| Q             | A
| ------------- | ---
| Is bugfix?    | no
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | no
| Fixed issues  | #14997

### CODE REVIEW

File: framework/db/Connection.php [hunk 1]
```diff
@@ -132,7 +132,7 @@
  * @author Qiang Xue <qiang.xue@gmail.com>
  * @since 2.0
  */
-class Connection extends Component
+class Connection extends Component implements ConnectionInterface
```
Comment (samdark) [inline framework/db/Connection.php]: It doesn't belong to this pull request.

Comment (njasm) [reply framework/db/Connection.php]: sure, I've branch it from the Connection one. I'll fix this.

**Comment by MysteryDragon** [comment]
There are not only UrlManagerInterface but also LoggerInterface. I suggest to adjust PR title. Or split PRs.

**Comment by samdark** [comment]
Splitting is better.

File: framework/db/Connection.php (hunk 1 above)
Comment (njasm) [reply framework/db/Connection.php]: @samdark rebased UrlManager into master parent, can you review if everything is ok? thanks!

**Comment by njasm** [comment]
removed ConnectionInterface commits from this pull request, created #15277 for that matter only.

**Review by samdark — APPROVED** [review]

File: framework/web/UrlManagerInterface.php [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+use yii\base\InvalidConfigException;
+
+/**
+ * UrlManagerInterface defines an handler for HTTP request parsing and creation of URLs based on a set of rules.
+ *
+ * UrlManager is configured as an application component in [[\yii\base\Application]] by default.
+ * You can access that instance via `Yii::$app->urlManager`.
+ *
+ * @author Nelson J Morais <njmorais@gmail.com>
```
Comment (SilverFire) [inline framework/web/UrlManagerInterface.php]: API was originally authored by Qiang. I think his authoring copyright should not be completely removed.

```
@author Qiang Xue <qiang.xue@gmail.com>
```

**Review by SilverFire — CHANGES_REQUESTED** [review]

**Review by sergeymakinen — APPROVED** [review]

File: framework/web/UrlManagerInterface.php (hunk 2 above)
Comment (njasm) [reply framework/web/UrlManagerInterface.php]: @SilverFire done, can you review it?

**Comment by cebe** [comment]
similar to https://github.com/yiisoft/yii2/pull/15277#issuecomment-383507354 the interface should require the minimal set of methods to be implemented, that are parseRequest, createUrl and createAbsoluteUrl in this case imo.
